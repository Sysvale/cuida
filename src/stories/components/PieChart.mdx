import { Meta, Story, Props, Controls, Canvas, Preview } from '@storybook/blocks';
import PieChart from '../../components/PieChart.vue';
import * as PieChartStories from './PieChart.stories';

<Meta of={PieChartStories} />

export const Template = (args) => ({
	components: { CdsPieChart: PieChart },
	setup() {
		return { args };
	},
	template: /*html*/ `
		<div style="background-color: #F8F9FA; padding: 16px; border-radius: 4px; height: 300px">
			<cds-pie-chart
				v-bind="args"
			/>
		</div>
	`
});

# PieChart

### Um componente de gráfico reutilizável para exibir dados visualmente.

***

⚠️ Esse componente vai ser substituído pelo DonutChart nas próximas versões do Cuida

<br />

## Quando usar:

* Exibir dados em formatos visuais;
* Visualizar tendências, proporções ou comparações em dados;
* Exibir proporções ou partes de um todo, onde cada fatia representa uma parcela do total;
* Funciona bem com categorias mutuamente exclusivas;

<br />

## Quando não usar:

* Para informações simples que podem ser exibidas em um formato tabular ou em texto;
* Quando os dados apresentarem muitas categorias ou subcategorias;
* Não é apropriado para dados contínuos ou dados temporais.

## Preview

<Canvas>
  <Story of={PieChartStories.PieChart} />
</Canvas>

<Controls />
