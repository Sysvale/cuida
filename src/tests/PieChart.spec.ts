import { describe, test, expect, vi } from 'vitest';
import PieChart from '../components/PieChart.vue';
import { shallowMount } from '@vue/test-utils';

vi.mock('../utils/methods/paleteBuilder.js', () => ({
	default: vi.fn((rawPalete) => {
		return [
			{
				colorName: 'Light Neutrals',
				variantName: 'Light',
				color400: '#E7EDF3',
				colorShades: ['#FFFFFF', '#FAFCFE', '#F4F8FB', '#E7EDF3', '#DFE5EC'],
				colorTokens: ['$n-0', '$n-10', '$n-20', '$n-30', '$n-40'],
				colorData: [
					{ shade: '#FFFFFF', token: '$n-0' },
					{ shade: '#FAFCFE', token: '$n-10' },
					{ shade: '#F4F8FB', token: '$n-20' },
					{ shade: '#E7EDF3', token: '$n-30' },
					{ shade: '#DFE5EC', token: '$n-40' }
				]
			},
			{
				colorName: 'Piccolo Green',
				variantName: 'Green',
				color400: '#6ddfb1',
				colorShades: ['#f3fcf8', '#e2f8ef', '#abedd3', '#6ddfb1', '#2db981'],
				colorTokens: ['$gp-50', '$gp-100', '$gp-200', '$gp-300', '$gp-400'],
				colorData: [
					{ shade: '#f3fcf8', token: '$gp-50' },
					{ shade: '#e2f8ef', token: '$gp-100' },
					{ shade: '#abedd3', token: '$gp-200' },
					{ shade: '#6ddfb1', token: '$gp-300' },
					{ shade: '#2db981', token: '$gp-400' }
				]
			}
		];
	})
}));

const mockedData =[
	{
		name: 'Ecocardiograma',
		datasets: [
			{
				label: 'Exames realizados',
				data: [120, 220, 180, 320, 150],
			},
		]
	},
];

const mockedColors = ['teal', 'violet', 'amber', 'blue', 'red'];

describe('PieChart', () => {
	test('renders correctly', async () => {
		const wrapper = shallowMount(PieChart, {
			props: {
				labels: ['Janeiro','Fevereiro','MarÃ§o', 'Abril', 'Maio'],
				variant: 'green',
				data: mockedData,
			},
		});

		expect(wrapper.html()).toMatchSnapshot();
	});

	test('it accepts valid colors', () => {
		const validator = PieChart.props.colors.validator;
		const invalidColors = [...mockedColors, 'silver'];
		expect(validator(mockedColors)).toBe(true);
		expect(validator(invalidColors)).toBe(false);
	});
});
