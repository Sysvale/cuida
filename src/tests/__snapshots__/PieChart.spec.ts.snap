// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`PieChart > renders correctly 1`] = `"<span data-v-218ded1e=""><div data-v-218ded1e="" class="responsive-container"><pie-stub data-v-218ded1e="" data="[object Object]" options="[object Object]" plugins="" datasetidkey="label"></pie-stub></div></span>"`;
